src/order_cmd.cpp

```diff
@@ -1517,6 +1517,40 @@ static bool CheckAircraftOrderDistance(const Aircraft *v_new, const Vehicle *v_o
 	return true;
 }
 
+/**
+ * Enable or disable automatic separation for a vehicle's order list
+ * @param tile unused
+ * @param flags operation to perform
+ * @param p1 VehicleIndex of the vehicle who's order list is being modified
+ * @param p2 value indicating whether to enable (1) or disable (0) automatic separation
+ * @param text unused
+ * @return the cost of this operation or an error
+ */
+CommandCost CmdOrderAutomaticSeparation(TileIndex tile, DoCommandFlag flags, uint32 p1, uint32 p2, const char *text)
+{
+	VehicleID veh = GB(p1, 0, 20);
+
+	Vehicle *vehicle = Vehicle::GetIfValid(veh);
+	if (vehicle == nullptr || !vehicle->IsPrimaryVehicle()) return CMD_ERROR;
+
+	CommandCost ret = CheckOwnership(vehicle->owner);
+	if (ret.Failed()) return ret;
+
+	if (flags & DC_EXEC) {
+		vehicle->SetAutomaticSeparationIsEnabled(p2);
```

Comment:
If automatic separation was previously enabled, this should probably call ResetAutomaticSeparation, as otherwise any old values left in `first_order_last_departure` will take effect and cause bugs.

Reply:
This code resets the data when you _disable_ automatic separation, and elsewhere we reset for every case which could cause issues with the data (e.g. when a vehicle is manually stopped).

I am happy to flip this to reset when you _enable_ automatic separation, or even in both directions, but I don't currently see how it would make any difference (outside of bugs elsewhere, which should be fixed in their own right).

Please let me know what I've missed here.